template.yaml

```diff
@@ -0,0 +1,368 @@
+          FromPort: 53
+          ToPort: 53
+          CidrIp: !Ref VpcCIDR
+
+        # DNS (53) UDP outbound
+        - IpProtocol: udp
+          FromPort: 53
+          ToPort: 53
+          CidrIp: !Ref VpcCIDR
+
+        # RDS PostgreSQL access
+        - IpProtocol: tcp
+          FromPort: !Sub "{{resolve:secretsmanager:${RDSConnectionSecretArn}:SecretString:port}}"
+          ToPort: !Sub "{{resolve:secretsmanager:${RDSConnectionSecretArn}:SecretString:port}}"
+          DestinationSecurityGroupId: !Ref RDSClientSGID
+
+  # ECS Resources
+  HapiFhirCluster:
+    Type: AWS::ECS::Cluster
+    Properties:
+      ClusterName: !Sub "${TestPrefix}${Env}hapi-fhir-cluster"
+      ClusterSettings:
+        - Name: containerInsights
+          Value: enabled
+
+  HapiFhirTaskDefinition:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      Family: !Sub "${TestPrefix}${Env}hapi-fhir-task"
+      NetworkMode: awsvpc
+      RequiresCompatibilities:
+        - FARGATE
+      Cpu: 2048
+      Memory: 4096
+      ExecutionRoleArn: !GetAtt ECSTaskExecutionRole.Arn
+      TaskRoleArn: !GetAtt ECSTaskRole.Arn
+      RuntimePlatform:
+        OperatingSystemFamily: LINUX
+        CpuArchitecture: ARM64
+      ContainerDefinitions:
+        - Name: !Sub "${TestPrefix}${Env}hapi-fhir-container"
+          Image: !Ref EcrImageUri
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-group: !Ref HapiFhirLogGroup
+              awslogs-region: !Ref AWS::Region
+              awslogs-stream-prefix: ecs
+          PortMappings:
+            - ContainerPort: 8080
+              HostPort: 8080
+              Protocol: tcp
+          Environment:
+            - Name: SPRING_DATASOURCE_URL
+              Value: !Join
+                - ""
+                - - "jdbc:postgresql://"
+                  - !Sub "{{resolve:secretsmanager:${RDSConnectionSecretArn}:SecretString:host}}"
+                  - ":"
+                  - !Sub "{{resolve:secretsmanager:${RDSConnectionSecretArn}:SecretString:port}}"
+                  - "/"
+                  - !Ref HapiFhirDBName
+            - Name: SPRING_DATASOURCE_USERNAME
+              Value: !Sub "{{resolve:secretsmanager:${RDSCredentialsSecretArn}:SecretString:username}}"
+            - Name: SPRING_DATASOURCE_PASSWORD
+              Value: !Sub "{{resolve:secretsmanager:${RDSCredentialsSecretArn}:SecretString:password}}"
+            - Name: POSTGRES_DB
+              Value: !Ref HapiFhirDBName
+            - Name: POSTGRES_USER
+              Value: !Sub "{{resolve:secretsmanager:${HapiFhirCredentialsSecretArn}:SecretString:username}}"
+            - Name: POSTGRES_PASSWORD
+              Value: !Sub "{{resolve:secretsmanager:${HapiFhirCredentialsSecretArn}:SecretString:password}}"
+
+  HapiFhirALB:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: !Sub "${TestPrefix}${Env}-hapi-fhir-alb"
+      Scheme: internal
```

Comment:
We think that the network load balancer is a better fit for this use case. The NLB offers better performance (and probably also price).  It can only route based on host:port, but that is all we need for HAPI-FHIR ECS.  Example:
   
```yaml
      Scheme: internal
      Type: network
      LoadBalancerAttributes: 
        - Key: load_balancing.cross_zone.enabled 
          Value: 'true'
        - Key: dns_record.client_routing_policy 
          Value: 'any_availability_zone'
```